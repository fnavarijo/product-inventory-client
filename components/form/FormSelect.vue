<template>
  <div>Select</div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'FormSelect',
});
</script>
