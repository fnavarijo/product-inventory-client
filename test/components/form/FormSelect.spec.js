import { mount } from '@vue/test-utils';

import FormSelect from '@/components/form/FormSelect.vue';

describe('FormSelect component', () => {
  it('should set the v-model value in the select', async () => {
    const wrapper = mount(FormSelect, {
      slots: [
        '<option value="desc">Descending</option>',
        '<option value="asc">Ascending</option>',
      ],
    });

    const selectInput = wrapper.find('select');
    await selectInput.setValue('asc');

    const selectedValue = selectInput.element.nodeValue;

    expect(selectedValue).toBe('asc');
  });

  it('should match snapshot', () => {
    const wrapper = mount(FormSelect, {
      slots: [
        '<option value="desc">Descending</option>',
        '<option value="asc">Ascending</option>',
      ],
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
