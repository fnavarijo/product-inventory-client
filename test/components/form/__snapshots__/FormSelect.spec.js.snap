// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelect component should match snapshot 1`] = `<select class="border border-gray-900"></select>`;
